utils/__tests__/validate_install_plans.test.js

```diff
@@ -30,44 +40,159 @@ const mockGithubAPIFiles = (filenames) =>
     patch: '',
   }));
 
+const mockGraphqlRequestBody = (variables = {}) => ({
+  queryString: VALIDATE_INSTALL_PLAN_MUTATION,
+  variables: {
+    id: 'infra-agent-targeted',
+    displayName: 'Test Install Agent',
+    heading: 'Test Install Install',
+    description: 'this is some description\n',
+    target: {
+      type: 'AGENT',
+      destination: 'HOST',
+      os: ['LINUX', 'WINDOWS'],
+    },
+    primary: {
+      mode: 'TARGETED',
+      destination: 'test-install-installer',
+    },
+    fallback: {
+      mode: 'LINK',
+      destination:
+        'https://docs.newrelic.com/docs/infrastructure/install-infrastructure-agent/linux-installation/install-infrastructure-monitoring-agent-linux/#manual-install',
+    },
+    ...variables,
+  },
+});
+
+const mockGraphqlResponse = ({ data = {}, errors } = {}) => ({
+  data,
+  ...(errors && { errors }),
+});
+
+const buildMockNRError = (message, argumentPath) => ({
+  message,
+  path: ['nr1CatalogSubmitInstallPlanStep'],
+  extensions: {
+    nerdGraphExtensions: { errorClass: 'VALIDATION_ERROR' },
+    argumentPath,
+  },
+});
+
 describe('Action: validate install plan id', () => {
   afterEach(() => {
     jest.resetAllMocks();
   });
 
-  test('succeeds with valid install plan id', () => {
-    const files = mockGithubAPIFiles([validQuickstartFilename]);
-    githubHelpers.filterQuickstartConfigFiles.mockReturnValueOnce(files);
+  test('succeeds with valid install plan', async () => {
+    const files = mockGithubAPIFiles([validInstallFilename]);
+    const requestBody = mockGraphqlRequestBody();
+    const response = mockGraphqlResponse();
+
+    githubHelpers.filterInstallPlans.mockReturnValueOnce(files);
+    nrGraphqlHelpers.fetchNRGraphqlResults.mockReturnValue(response);
+
+    await validateInstallPlan(files);
```

Comment:
Would it be worth asserting on the returned boolean here?
```suggestion
   const hasFailed = await validateInstallPlan(files);
```

Suppose the value would be: can we throw / log errors and still have the `process.exit` run with the default `0` argument—giving a false positive. If so, might as well assert on the return here to make sure that final block is being  addressed as we would hope. 

```
  if (hasFailed) {
    process.exit(1);
  }
```

Suppose you could could also `spyOn` the `process.exit` function if that feels more of a sure thing

Reply:
very true!!! good catch!